src/utils.ts

```diff
@@ -222,3 +223,64 @@ export async function retry<T>(call: () => Promise<T>, times: number, delayS: nu
     });
   return promiseChain;
 }
+
+/**
+ * Estimates the total gas cost required to submit an unsigned (populated) transaction on-chain
+ * @param unsignedTx The unsigned transaction that this function will estimate
+ * @param relayerAddress The address that the transaction will be submitted from ()
+ * @param provider A valid ethers provider - will be used to reason the gas price
+ * @param gasPrice A manually provided gas price - if set, this function will not resolve the current gas price
+ * @returns The total gas cost to submit this transaction - i.e. gasPrice * estimatedGasUnits
+ */
+export async function estimateTotalGasRequiredByUnsignedTransaction(
+  unsignedTx: PopulatedTransaction,
+  relayerAddress: string,
```

Comment:
tiny nit: nothing else in this function seems specific to a relay, so you might consider changing this variable name to senderAddress since it could be used for _any_ transaction.

Reply:
That makes total sense! I've updated it [here](https://github.com/across-protocol/sdk-v2/pull/64/commits/451c80e04daffbffb6d2999e873731a4b4c46ba3).